doc/guides/coverage.rst

```diff
@@ -44,25 +44,39 @@ To report the coverage for the particular test application set :option:`CONFIG_C
 Steps to generate code coverage reports
 =======================================
 
-1. Build the code with CONFIG_COVERAGE=y::
+These steps will produce an HTML coverage report for a single application.
+
+1. Build the code with CONFIG_COVERAGE=y. Some boards like qemu_x86_coverage
+   automatically enable this, but for boards that do not you will need to
+   enable the configuration manually::
 
      $ cmake -DBOARD=mps2_an385 -DCONFIG_COVERAGE=y ..
+     $ make
 
-#. Store the build and run output on to a log file::
+#. Capture the emulator output into a log file. You may need to terminate
+   the emulator with ``Ctrl-A X`` for this to complete after the coverage dump
+   has been printed::
 
-     $ make run > log.log
+     $ make run | tee log.log
 
-#. Generate the gcov gcda files from the log file that was saved::
+#. Generate the gcov ``.gcda`` and ``.gcno`` files from the log file that was
+   saved::
 
      $ python3 scripts/gen_gcov_files.py -i log.log
 
-#. Find the gcov binary placed in the SDK::
+#. Find the gcov binary placed in the SDK. You will need to pass the path to
+   the gcov binary for the appropriate architecture when you later invoke
+   ``gcovr``::
+
+     $ find $ZEPHYR_SDK_INSTALL_DIR -iregex ".*gcov"
 
-     $ find -iregex ".*gcov"
+#. Create an output directory for the reports::
 
-#. Run gcovr to get the reports::
+     $ mkdir -p gcov_report
 
-     $ gcovr -r . --html -o gcov_report/coverage.html --html-details --gcov-executable <gcov_path_in_SDK>
+#. Run ``gcovr`` to get the reports::
+
+     $ gcovr -r $ZEPHYR_BASE zephyr --html -o gcov_report/coverage.html --html-details --gcov-executable <gcov_path_in_SDK>
```

Comment:
remove that extra zephyr

Reply:
@nashif no.

That zephyr tells gcovr where to look for gcda/gcno files. It should not be searching for them in $ZEPHYR_BASE.

Reply:
usage: gcovr [options] [search_paths...]

Reply:
```
Traceback (most recent call last):
  File "/usr/bin/gcovr", line 11, in <module>
    load_entry_point('gcovr==4.1', 'console_scripts', 'gcovr')()
  File "/usr/lib/python3.7/site-packages/gcovr/__main__.py", line 548, in main
    datafiles = get_datafiles(options.search_paths, options)
  File "/usr/lib/python3.7/site-packages/gcovr/gcov.py", line 53, in get_datafiles
    ".*\.gc(da|no)$", dir_, exclude_dirs=options.exclude_dirs)
  File "/usr/lib/python3.7/site-packages/gcovr/utils.py", line 153, in search_file
    raise IOError("Unknown directory '" + path + "'")
OSError: Unknown directory 'zephyr'
```

Reply:
i noticed you changed it to "." now...

Reply:
@nashif these instructions are from the out directory, of course zephyr/ exists